main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -669,7 +669,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     }
 
     def LetMatch: Rule1[ParsedAst.Expression.LetMatch] = rule {
-      SP ~ keyword("let") ~ WS ~ Pattern ~ optWS ~ optional(":" ~ optWS ~ Type ~ optWS) ~ "=" ~ optWS ~ Expression ~ optWS ~ ";" ~ optWS ~ Stm ~ SP ~> ParsedAst.Expression.LetMatch
+      SP ~ keyword("let") ~ WS ~ Modifiers ~ Pattern ~ optWS ~ optional(":" ~ optWS ~ Type ~ optWS) ~ "=" ~ optWS ~ Expression ~ optWS ~ ";" ~ optWS ~ Stm ~ SP ~> ParsedAst.Expression.LetMatch
```

Comment:
I think this allows `let scopedx` ?

Reply:
What happens to:

```
let scoped (x, y) = ...
```

Does that work?

If it does, should we also allow scoped in patterns?

Reply:
`scoped` is wrapped with `keyword` which prevents letters from appearing directly afterward

The check in the `Weeder` only allows `scoped` on patterns that are just a var. This
```
let scoped (x, y) = ...
```
throws an `IllegalModifier`

I think we can allow it in patterns in a future improvement.